mvp_example/src/main/java/main/presenter/PresenterImplementation.kt

```diff
@@ -72,12 +72,10 @@ class PresenterImplementation(
                 .catch {
                     withContext(Dispatchers.Main) {
                         when (this@catch) {
-                            is IllegalArgumentException -> {
-                                Log.d(TAG, "Cheating death!")
-                            }
-                            else -> {
-                                Log.d(TAG, "Unknown exception")
-                            }
+                            is IllegalArgumentException ->
+                                view.showError("Cheating death!")
```

Comment:
minor: 
```suggestion
                            is IllegalArgumentException ->  view.showError("Cheating death!")
```
one liners are better for readability, or return back brackets instead
another way is 
```
val errorType = when (this@catch) {
    is IllegalArgumentException -> CHEATING_ERROR
    else -> GENERAL_ERROR
}
view.showError(errorType)
```

Reply:
note that returning something from when makes it exhaustive, which is more future-proof in case you are not using `else` as it will be a compiler error when adding more `when` branches

Reply:
Changed it to your propposal, Indeed iw more future-proof
Thanks!